### PR TITLE
Beta server stub generator for F#/Giraffe

### PR DESCRIPTION
**Pull request by nmfisher**
### PR checklist

-  [✓] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [✓] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh`, `./bin/openapi3/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`~~, `3.4.x`, `4.0.x`~~. Default: `master`.
- [ ] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

This is a beta server stub generator for F#/Giraffe as discussed in #2342. 

Much of the OpenAPI specification is still unsupported, but this is mature enough to open up to the community for discussion/comments/contributions.

### CODE REVIEW

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractFSharpCodegen.java [hunk 1]
```diff
@@ -0,0 +1,1166 @@
+/*
+ * Copyright 2018 OpenAPI-Generator Contributors (https://openapi-generator.tech)
+ * Copyright 2018 SmartBear Software
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractFSharpCodegen.java]: @nmfisher please remove line 3 as per the discussion in https://opensource.stackexchange.com/questions/7300/copyright-notice-in-the-file-header-apache-v2-license/7301#7301

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/FsharpGiraffeServerCodegen.java [hunk 2]
```diff
@@ -0,0 +1,288 @@
+/*
+ * Copyright 2018 OpenAPI-Generator Contributors (https://openapi-generator.tech)
+ * Copyright 2018 SmartBear Software
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/FsharpGiraffeServerCodegen.java]: @nmfisher please remove line 3 as per the discussion in https://opensource.stackexchange.com/questions/7300/copyright-notice-in-the-file-header-apache-v2-license/7301#7301

Comment (nmfisher) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/FsharpGiraffeServerCodegen.java]: Done.

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractFSharpCodegen.java (hunk 1 above)
Comment (nmfisher) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractFSharpCodegen.java]: Done.

**Comment by wing328** [comment]
@nmfisher FYI. I've removed carriage return in the template files via https://github.com/OpenAPITools/openapi-generator/pull/2705/commits/bdebd81c136e9013824c1504acc23cafb452dc21 (I also make some minor enhancements to the shell script)

File: samples/server/petstore/fsharp-giraffe/src/OpenAPI.fsproj [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    <PackageReference Include="TaskBuilder.fs" Version="2.1.*" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <Content Include="openapi.yaml"/>
+    <Compile Include="Model.fs" />
+    <Compile Include="api/params/UrlParams.fs" />
+    <Compile Include="api/params/QueryParams.fs" />
+    <Compile Include="api/params/BodyParams.fs" />
+    <Compile Include="api/params/HeaderParams.fs" />
```
Comment (wing328) [inline samples/server/petstore/fsharp-giraffe/src/OpenAPI.fsproj]: @nmfisher Looks like these 4 files under `api/params` are not yet added to the project (templates)

**Comment by nmfisher** [comment]
Sorry about the CRLF - will configure git/IDE to stick with LF going forward.

**Comment by wing328** [comment]
@nmfisher no worries. We've a script to catch these CRLF as part of the CI workflow.

File: samples/server/petstore/fsharp-giraffe/src/OpenAPI.fsproj (hunk 3 above)
Comment (nmfisher) [reply samples/server/petstore/fsharp-giraffe/src/OpenAPI.fsproj]: Actually that directory (samples/server/petstore/fsharp-giraffe/src) was the sample output directory from an earlier revision.

The proper output directory is samples/server/petstore/fsharp-giraffe/{{packageName}} (e.g. samples/server/petstore/fsharp-giraffe/OpenAPI)

I just forgot to remove the earlier version (samples/server/petstore/fsharp-giraffe/src) from the final commit. This is OK to delete - see https://github.com/nmfisher/openapi-generator/commit/ee4672ecf1fc6060e351797fe692736da4a4fc01

**Comment by wing328** [comment]
OK. I'll test again.

If no more question from me, I'll merge it into the master to release a beta version so as to collect feedback from the community.

**Comment by wing328** [comment]
Tests passed via https://travis-ci.org/OpenAPITools/openapi-generator/builds/527636181.

The CircleCI failure is due to the outdated doc. I'll fix it with another PR after merging this into master.
Travis failure due to connection timeout (not related to this change):
>  Connect to repo.maven.apache.org:443 [repo.maven.apache.org/151.101.184.215] failed: Read timed out

**Pull request merged**